rpc/server.go

```diff
@@ -45,6 +45,14 @@ type serverResponse struct {
 	err   error
 }
 
+type syncAdapter func(proto.Message) (proto.Message, error)
```

Comment:
Maybe; I don't feel strongly either way. I'm not happy with the use of "sync" and "async" here in general: is a handler that takes a callback asynchronous because it may not finish until some time after it has returned, or is it synchronous because it is called synchronously with respect to the rpc server's thread?